service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -315,6 +317,8 @@ public void close() throws IOException
             throw new IOException("AMQP sender is not initialized");
         }
 
+        this.executor.shutdownNow();
```

Comment:
One other option here is to do ```this.executor.shutdown()``` instead. That approach allows threads to finish "gracefully" but our AMQP handlers aren't really setup to handle an interrupt in the first place so that seemed unnecessary to me.